bot/bot.go

```diff
@@ -32,42 +32,95 @@ type Bot interface {
 	Channels() []string                                       // Channels returns list of channels which bot has joined.
 }
 
-// NewBot creates a new Bot implementation with a set of parameters.
-func NewBot(host string, ssl bool, nick, ident string, channels []string) Bot {
-	return NewBotWithProxy(host, ssl, nick, ident, channels, "")
-}
-
-// NewBotWithProxy creates a new Bot implementation with a set of parameters and a proxy.
-func NewBotWithProxy(host string, ssl bool, nick, ident string, channels []string, proxy string) Bot {
+// NewBotWithProxyPassword creates a new Bot implementation with required parameters and options.
+func NewBotOptions(host string, nick string, options ...func(*BotImpl)) (Bot, error) {
 	hostPort := strings.SplitN(host, ":", 2)
 	cfg := &client.Config{
-		Me:          &state.Nick{Nick: nick},
+		Me: &state.Nick{
+			Nick:  nick,
+			Ident: nick,
+			Name:  nick,
+		},
 		NewNick:     func(s string) string { return s + "_" },
 		PingFreq:    3 * time.Minute,
 		QuitMessage: "I have to go.",
 		Server:      host,
-		SSL:         ssl,
+		SSL:         false,
 		SSLConfig:   tls.Config(hostPort[0]),
 		Version:     "Powered by GoIRCBot",
 		Recover:     (*client.Conn).LogPanic,
 		SplitLen:    450,
-		Proxy:       proxy,
+		Proxy:       "",
+		Pass:        "",
 	}
-	cfg.Me.Ident = ident
-	cfg.Me.Name = nick
 
-	conn := client.Client(cfg)
-	conn.EnableStateTracking()
 	b := &BotImpl{
-		conn:      conn,
+		config:    cfg,
 		reconnect: true,
 		quit:      make(chan bool),
 		commands:  NewCommands(),
-		channels:  channels,
 	}
 
+	for _, option := range options {
+		option(b)
+	}
+
+	return b, nil
```

Comment:
So we don't use errors today, but you want to avoid another API breakage if we ever need errors in initialization, is that right? I don't like unnecessary things, which may remain forever so, but it's also not a very big pain for callers to check it when they initialize, so I guess it's fine.

Reply:
Yes that's my thinking. Especially since `Run()` doesn't return err